sources/HUBViewControllerImplementation.m

```diff
@@ -609,11 +610,18 @@ - (void)reloadCollectionViewWithViewModel:(id<HUBViewModel>)viewModel animated:(
     /* Performing batch updates inbetween viewDidLoad and viewDidAppear is seemingly not allowed, as it
      causes an assertion inside a private UICollectionView method. If no diff exists, fall back to
      a complete reload. */
-    if (!self.viewHasAppeared || self.lastViewModelDiff == nil) {
+    if (!self.hasPerformedInitialReload || !self.viewHasAppeared || self.lastViewModelDiff == nil) {
         [self.collectionView reloadData];
         
         [layout computeForCollectionViewSize:self.collectionView.frame.size viewModel:viewModel diff:self.lastViewModelDiff];
+        
+        /* Forcing a re-layout as the reloadData-call doesn't trigger the numberOfItemsInSection:-calls 
+           by itself, and batch update calls don't play well without having an initial item count. */
```

Comment:
Actually I was able to reproduce the crash even after the `numberOfItemsInSection:` was called initially so not sure that this guarantees the expected behaviour.